web/src/message_list_view.js

```diff
@@ -1250,7 +1250,9 @@ export class MessageListView {
         this._post_process($rendered_msg);
         $row.replaceWith($rendered_msg);
 
-        if (was_selected) {
+        if (was_selected && this.list === message_lists.current) {
+            // We don't need to select the message if this message list is not rendered as
+            // doing so causes the message to be marked as read unintentionally.
```

Comment:
I can't understand the issue here -- the reproduction recipe seems to work for me with this change. 

Perhaps you can open this in a new PR with a more detailed description showing the bug; I'm going to merge the other change closing this PR since it's a thing I understand and know is a critical release goal, and it'll be nice to have this second issue gets it's own tracking identity.